### PR TITLE
Add essential maven packages

### PR DESCRIPTION
**Pull request by rikenm1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Change: Add javapackages-bootstrap and essential maven packages and plugins

###### Does this affect the toolchain?  

**YES/NO**
No

###### Test Methodology

- Pipeline build id: [333689](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=333689&view=results)
- Pipeline build id: [333692](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=333692&view=results)

### CODE REVIEW

File: SPECS/javapackages-bootstrap/javapackages-bootstrap.spec [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+Name:           javapackages-bootstrap
+Version:        1.5.0
+Release:        2%{?dist}
+Summary:        A means of bootstrapping Java Packages Tools
+# For detailed info see the file javapackages-bootstrap-PACKAGE-LICENSING
+License:        ASL 2.0 and ASL 1.1 and (ASL 2.0 or EPL-2.0) and (EPL-2.0 or GPLv2 with exceptions) and MIT and (BSD with advertising) and BSD-3-Clause and EPL-1.0 and EPL-2.0 and CDDL-1.0 and xpp and CC0 and Public Domain
+URL:            https://github.com/fedora-java/javapackages-bootstrap
+BuildArch:      noarch
+
+Source0:        https://github.com/fedora-java/javapackages-bootstrap/releases/download/%{version}/javapackages-bootstrap-%{version}.tar.xz
```
[the remark this answers is not in the document]
Comment (nicogbg) [reply SPECS/javapackages-bootstrap/javapackages-bootstrap.spec:23]: if you go to the script please name it "generate_source_tarball.sh" (See https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/mariadb/generate_source_tarball.sh)

File: SPECS/maven-resources-plugin/maven-resources-plugin.spec [hunk 2]
```diff
@@ -0,0 +1,206 @@
+
+Summary:        Maven Resources Plugin
+Name:           maven-resources-plugin
+Version:        3.2.0
```
Comment (nicogbg) [inline SPECS/maven-resources-plugin/maven-resources-plugin.spec:4]: why not using the latest (3.3.1) ?

File: SPECS/maven-compiler-plugin/maven-compiler-plugin.spec [hunk 3]
```diff
@@ -0,0 +1,222 @@
+%bcond_without bootstrap
+Summary:        Maven Compiler Plugin
+Name:           maven-compiler-plugin
+Version:        3.8.1
```
Comment (nicogbg) [inline SPECS/maven-compiler-plugin/maven-compiler-plugin.spec:4]: why not using latest (3.11.0)?

File: SPECS/javapackages-bootstrap/javapackages-bootstrap.spec [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+%global         __requires_exclude_from \\.jar$
+%global         __provides_exclude_from \\.jar$
+
+%global mavenHomePath %{_datadir}/%{name}
+%global metadataPath %{mavenHomePath}/maven-metadata
+%global artifactsPath %{_javadir}
+%global launchersPath %{_libexecdir}/%{name}
+
+Name:           javapackages-bootstrap
+Version:        1.5.0
```
Comment (nicogbg) [inline SPECS/javapackages-bootstrap/javapackages-bootstrap.spec:15]: why not using latest (1.9.0) ? 1.5.0 is from 07/2021

File: SPECS/google-guice/google-guice.spec [hunk 5]
```diff
@@ -0,0 +1,479 @@
+%bcond_without bootstrap
+Summary:        Lightweight dependency injection framework for Java 5 and above
+Name:           google-guice
+Version:        4.2.3
```
Comment (nicogbg) [inline SPECS/google-guice/google-guice.spec:4]: why not using latest (5.1.0)? 4.2.3 is from 03/2020

File: SPECS/maven-surefire/maven-surefire.signatures.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+    "Signatures": {
+     "maven-surefire-3.0.0.tar.gz": "3a8ab67869ac4b83cde22ca23b68140b2b46feb7d970db95e42923eb95f9f908",
+     "cpl-v10.html":"dde4c2b665c81a60a6dec39beb54b89836ae071e552d935a04880f9d5a2e58cd",
+     "generate-tarball.sh":"b223db29e37a7bfd28678bc01b63131e28158e97568ca8d3a4d870256d0bddcf"
```
Comment (nicogbg) [inline SPECS/maven-surefire/maven-surefire.signatures.json:5]: do you need to embed generate-tarball.sh in package ? if not I don't think we need its signature

File: SPECS/maven-surefire/maven-surefire.spec [hunk 7]
```diff
@@ -0,0 +1,515 @@
...
+Version:        3.0.0
+Release:        1%{?dist}
+License:        ASL 2.0 AND CPL
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://maven.apache.org/surefire/
+# ./generate-tarball.sh
+Source0:        %{_mariner_sources_url}/%{name}-%{version}.tar.gz
+# Remove bundled binaries which cannot be easily verified for licensing
+Source1:        generate-tarball.sh
```
Comment (nicogbg) [inline SPECS/maven-surefire/maven-surefire.spec:14]: this file does not seem to be used

**Review by nicogbg — CHANGES_REQUESTED** [review]

File: SPECS/maven-surefire/maven-surefire.spec (hunk 7 above)
Comment (rikenm1) [reply SPECS/maven-surefire/maven-surefire.spec:14]: generate-tarball.sh is used to strip jar files from the source.

File: SPECS/maven-resources-plugin/maven-resources-plugin.spec (hunk 2 above)
Comment (rikenm1) [reply SPECS/maven-resources-plugin/maven-resources-plugin.spec:4]: bootstrap supports 3.2.0.
The build fails with newer versions.

File: SPECS/maven-resolver/maven-resolver.spec [hunk 8]
```diff
@@ -0,0 +1,230 @@
+
+Summary:        Apache Maven Artifact Resolver library
+Name:           maven-resolver
+Version:        1.7.3
```
[the remark this answers is not in the document]
Comment (rikenm1) [reply SPECS/maven-resolver/maven-resolver.spec:4]: same reason as above

File: SPECS/maven-jar-plugin/maven-jar-plugin.spec [hunk 9]
```diff
@@ -0,0 +1,206 @@
+%bcond_without bootstrap
+Summary:        Maven JAR Plugin
+Name:           maven-jar-plugin
+Version:        3.2.2
```
[the remark this answers is not in the document]
Comment (rikenm1) [reply SPECS/maven-jar-plugin/maven-jar-plugin.spec:4]: bootstrap supports 3.2.2.
The build fails with newer versions.

File: SPECS/maven-compiler-plugin/maven-compiler-plugin.spec (hunk 3 above)
Comment (rikenm1) [reply SPECS/maven-compiler-plugin/maven-compiler-plugin.spec:4]: bootstrap supports 3.8.1.
The build fails with newer versions.

File: SPECS/maven-wagon/maven-wagon.spec [hunk 10]
```diff
@@ -0,0 +1,365 @@
+%bcond_without bootstrap
+Summary:        Tools to manage artifacts and deployment
+Name:           maven-wagon
+Version:        3.5.1
```
[the remark this answers is not in the document]
Comment (rikenm1) [reply SPECS/maven-wagon/maven-wagon.spec:4]: bootstrap supports 3.5.1.
The build fails with newer versions.

File: SPECS/javapackages-bootstrap/javapackages-bootstrap.spec (hunk 4 above)
Comment (rikenm1) [reply SPECS/javapackages-bootstrap/javapackages-bootstrap.spec:15]: this is the version where fedora was using openjdk-11. 
the newer version uses openjdk-17. Hence had to use the older version.

File: SPECS/google-guice/google-guice.spec (hunk 5 above)
Comment (rikenm1) [reply SPECS/google-guice/google-guice.spec:4]: bootstrap supports 4.2.3.
The build fails with newer versions.

File: SPECS/maven-surefire/generate-tarball.sh [hunk 11]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (rikenm1) [reply SPECS/maven-surefire/generate-tarball.sh:1]: I can. update all the script files to match the format we had. I have seen some other java spec inside mariner with this same generate_tarball.sh script file where all it does is remove the jar file from the source tar. 

I did also see in fedora they would remove jar files from the source tar in %prep sometimes.  

For me, I think removing those jar files in %prep is fine but the source which we have in the blob storage will have the jar.  
I am fine either way. your input would be awesome on this.

File: SPECS/google-guice/create-tarball.sh [hunk 12]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (rikenm1) [reply SPECS/google-guice/create-tarball.sh]: resolved

File: SPECS/maven-surefire/maven-surefire.signatures.json (hunk 6 above)
Comment (rikenm1) [reply SPECS/maven-surefire/maven-surefire.signatures.json]: removed. thanks for pointing it out. Fedora folks had this way.

File: SPECS/maven-surefire/generate-tarball.sh (hunk 11 above)
[the remark this answers is not in the document]
Comment (rikenm1) [reply SPECS/maven-surefire/generate-tarball.sh]: updated.

File: SPECS/maven-surefire/maven-surefire.spec (hunk 7 above)
Comment (rikenm1) [reply SPECS/maven-surefire/maven-surefire.spec]: removed as not needed to be embedded

**Review by nicogbg — APPROVED** [review]

**Pull request merged**